### PR TITLE
Delete the takes package if all takes are spiked.

### PR DESCRIPTION
**Pull request by mdhaman**

### CODE REVIEW

File: server/apps/packages/package_service.py [hunk 1]
```diff
@@ -257,20 +257,26 @@ def remove_refs_in_package(self, package, ref_id_to_remove, processed_packages=N
...
-        insert_into_versions(id_=package[config.ID_FIELD])
+            if new_sequence == 0:
+                get_resource_service(ARCHIVE).delete_action({config.ID_FIELD: package[config.ID_FIELD]})
+                delete_package = True
+            else:
+                updates[SEQUENCE] = new_sequence
+                last_take_group = next(reference for reference in
+                                       next(new_group.get('refs') for new_group in new_groups if
+                                            new_group['id'] == 'main')
+                                       if reference.get(SEQUENCE) == new_sequence)
```
Comment (plamut) [inline server/apps/packages/package_service.py]: Woohoo, generators, good job! :)

**Comment by plamut** [comment]
:+1:

**Comment by sivakuna-aap** [comment]
:+1: Need to improve the doc string of the method. Also place the doc string in `line 188` under the appropriate method.

**Comment by akintolga** [comment]
:+1:

**Pull request merged**